structuredheadings/plugin.js

```diff
@@ -546,15 +545,15 @@
         return false;
       };
 
-      var headings = [];
-      var current = walker.next();
-
-      while (current) {
-        headings.push(current);
-        current = walker.next();
+     var filteredIterator = function () {
```

Comment:
My reading is that this returns either the first heading or the last paragraph (if no heading). So maybe a still-poor but more descriptive name might be `firstHeadingOrLastParagraph`?